test/libsolidity/memoryGuardTests/free_function.sol

```diff
@@ -0,0 +1,29 @@
+function safe() pure returns (uint256 x) {
+    assembly { x := 42 }
+    /// @solidity memory-safe-assembly
+    assembly { mstore(0, 0) }
+}
+function unsafe() pure returns (uint256 x) {
+    assembly { pop(mload(0)) }
```

Comment:
Not sure reading actually needs to be unsafe...

Reply:
We can relax it later, but I don't see why you would read if you don't write...